xbmc/settings/SubtitlesSettings.cpp

```diff
@@ -8,14 +8,48 @@
 
 #include "SubtitlesSettings.h"
 
+#include "FileItem.h"
 #include "ServiceBroker.h"
+#include "Util.h"
+#include "filesystem/Directory.h"
+#include "filesystem/File.h"
 #include "settings/Settings.h"
 #include "settings/SettingsComponent.h"
 #include "settings/lib/Setting.h"
+#include "utils/FontUtils.h"
+#include "utils/StringUtils.h"
+#include "utils/URIUtils.h"
+
+#include <algorithm>
+#include <mutex>
+#include <thread>
+#include <vector>
 
 using namespace KODI;
 using namespace SUBTITLES;
 
+namespace
+{
+void addFont(std::mutex& vecFontMutex,
+             const CFileItemPtr item,
+             std::vector<std::string>& fontCollection)
+{
+  std::string filename = item->GetLabel();
+  // Font filename that start with "tmp.font." are excluded because
+  // are temporary fonts extracted by the demuxer from MKV files
+  if (!item->m_bIsFolder && CUtil::IsSupportedFontExtension(filename) &&
```

Comment:
Maybe move to font utils too to wrap the comparison and create some single point of access?

Reply:
Thinking a bit more about this, since now we have support for `ass_add_font` and we got rid of the support for two font folds. We can probably get rid of this and make the demuxer extract fonts to the temp folder just like before.
Fonts in temp/Fonts will then be added on initialization using `ass_add_font`. Not mandatory to change then, I can do a followup to get rid of this once the PR gets in

Reply:
I have just made this requested change locally
it does not matter even if my changes are deleted immediately afterwards,
however that would be wonderful